### ISSUE TITLE
WinUI Packaged projects throws Markup.XamlParseException: 'XAML parsing failed.' 

### ISSUE THREAD

**Issue by strfast365** [microsoft/microsoft-ui-xaml#6452]
### Describe the bug

Single-project MSIX throws Markup.XamlParseException: 'XAML parsing failed.'  when using templated control from ClassLibrary referenced by dll.

### Steps to reproduce the bug

Steps to reproduce:
1) Create WinUI class library
2) Add empty custom control to the project
3) Create a new Single-project MSIX project
4) Add a new reference to dll of the class library
5) Add custom control from class library to MainWindow.xaml
6) Run project

[WinUIRes.zip](https://github.com/microsoft/microsoft-ui-xaml/files/7684132/WinUIRes.zip)

### Expected behavior

You can also create a new solution with a separate packaging project and repeat step 3,4,5,6 - it works fine

### Windows app type

- [ ] UWP
- [x] Win32

### Device form factor

Desktop

### Windows version

May 2021 Update (19043)

**Comment by StephenLPeters** [comment]
@RealTommyKlein and @evelynwu-msft FYI

**Comment by sam-wheat** [comment]
What is the status on this ticket please?  Suggested workaround does not work - are there any others?

Thank you.

**Comment by dotMorten** [comment]
I don’t think this could work. You need to use either a project reference or pack your output as a nuget package and reference that to get the resources as well. The DLL itself doesn’t carry enough in itself. This was also a requirement in the UWP days

**Comment by sam-wheat** [comment]
> or pack your output as a nuget package and reference that to get the resources as well.

@dotMorten  In my case my control works fine when referenced as a project however when I create a nuget package and reference that instead I get "XAML parsing failed".  

Why would it not work?  Seems like the standard use case for nuget packages.

**Comment by dotMorten** [comment]
@sam-wheat see the issue linked a few messages up. It takes a bit of hacking to get it to make a usable package

**Comment by Felix-CodingClimber** [comment]
Same problem here...

**Comment by pmmachado1971** [comment]
Hello. Same problem here. Anyone have a solution?
I need to develop and App that loads plugins that have WinUI 3 pages/windows and always get xaml parsing failed is dll loaded externely. If class library is referenced directly from the project everything works ok, but I need to allow others to develop plugins with user interface and load the dynamically.
Any help will be appreciated.

**Comment by haonanttt** [comment]
I've created a discussion thread giving the workaround about how to solve the 'XAML parsing failed.' issue in this case. The link is https://github.com/microsoft/microsoft-ui-xaml/discussions/10268.

In short, when generating the nuget package, we could either:

1. Set DisableEmbeddedXbf property to false when packing a 'Debug' nuget package
or
2. Pack a 'Release' nuget package instead

The workaround is same to what @jeromelaban had mentioned, and in a similar issue https://github.com/microsoft/microsoft-ui-xaml/issues/7830 I've tried it worked

**Comment by VishalOmprasad** [comment]
Hi Team,

Checked with this suggested workaround: [https://github.com/microsoft/microsoft-ui-xaml/issues/6452#issuecomment-2568935977](https://github.com/microsoft/microsoft-ui-xaml/issues/6452#issuecomment-2568935977), however the issue still replicated.

Is there any insight or workaround to resolve this issue? Any help or update on the progress would be greatly appreciated.
